### PR TITLE
Add config for endpoint

### PR DESCRIPTION
**Pull request by ChristianKniep**
Provides a way to specify the endpoint in a config file.
If the endpoint is not set the default will use the unix socket.
```
cat /etc/fullerite/conf.d/DockerStats.conf
{}
```
If it's false, it fails...
```
bash-4.3# cat /etc/fullerite/conf.d/DockerStats.conf                          f
{
   "dockerEndPoint": "unix:///var/run/docker.FALSE"
}
bash-4.3# /fullerite/build/usr/bin/fullerite -c /etc/fullerite/fullerite.conf
time="04 Mar 16 19:37 UTC" level=info msg="Starting fullerite..." app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Reading configuration file at /etc/fullerite/fullerite.conf" app=fullerite pkg=config
time="04 Mar 16 19:37 UTC" level=info msg="Starting collectors..." app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Reading collector configuration file at /etc/fullerite/conf.d/DockerStats.conf" app=fullerite pkg=config
time="04 Mar 16 19:37 UTC" level=info msg="Starting handlers..." app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Starting handler Graphite" app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Starting to run internal metrics server on port 29090 on path /metrics" app=fullerite pkg=internalserver
time="04 Mar 16 19:37 UTC" level=info msg="Running DockerStatsCollector" app=fullerite
time="04 Mar 16 19:37 UTC" level=error msg="ListContainers() failed: Get http://unix.sock/containers/json: dial unix /var/run/docker.FALSE: connect: no such file or directory" app=fullerite collector=DockerStats pkg=collector
bash-4.3#
```
correct leads to success... :)
```
bash-4.3# cat /etc/fullerite/conf.d/DockerStats.conf                          f
{
   "dockerEndPoint": "unix:///var/run/docker.sock"
}
bash-4.3# /fullerite/build/usr/bin/fullerite -c /etc/fullerite/fullerite.conf
time="04 Mar 16 19:37 UTC" level=info msg="Starting fullerite..." app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Reading configuration file at /etc/fullerite/fullerite.conf" app=fullerite pkg=config
time="04 Mar 16 19:37 UTC" level=info msg="Starting collectors..." app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Reading collector configuration file at /etc/fullerite/conf.d/DockerStats.conf" app=fullerite pkg=config
time="04 Mar 16 19:37 UTC" level=info msg="Starting handlers..." app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Starting handler Graphite" app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Starting to run internal metrics server on port 29090 on path /metrics" app=fullerite pkg=internalserver
time="04 Mar 16 19:37 UTC" level=info msg="Running DockerStatsCollector" app=fullerite
time="04 Mar 16 19:37 UTC" level=info msg="Starting to emit 18 metrics" app=fullerite handler=Graphite pkg=handler
time="04 Mar 16 19:37 UTC" level=info msg="POST of 18 metrics to Graphite took 0.011235 seconds" app=fullerite handler=Graphite pkg=handler
bash-4.3#
```

### CODE REVIEW

File: src/fullerite/collector/docker_stats.go [hunk 1]
```diff
@@ -69,6 +69,18 @@ func (d *DockerStats) Configure(configMap map[string]interface{}) {
 	} else {
 		d.statsTimeout = d.interval
 	}
+	if dockerEndpoint, exists := configMap["dockerEndPoint"]; exists {
+		if str, ok := dockerEndpoint.(string); ok {
+			d.endpoint = str
+		} else {
+    		etype := fmt.Sprintf("%T", dockerEndpoint)
```
Comment (gnufied) [inline src/fullerite/collector/docker_stats.go]: indentation problem?

Comment (gnufied) [inline src/fullerite/collector/docker_stats.go]: I would perhaps use `reflect.TypeOf(dockerEndpoint)` in log rather than doing this in two steps.

**Comment by tsheasha** [comment]
+1 lgtm

**Comment by gnufied** [comment]
+1 please fix test and we can merge this.

**Comment by ChristianKniep** [comment]
Hey @gnufied, @tsheasha,

I am afraid I am confused with the way how to test the case. Could you guys give me a hand in the test?

Cheers
Christian

**Comment by gnufied** [comment]
@ChristianKniep this is what we meant - https://github.com/Yelp/fullerite/commit/0c73f6e84ea62bc74ac9fc09df5eb292a83d78fc

**Comment by ChristianKniep** [comment]
OK, this shoudl do it then...
```
bash-4.3# make
Cleaning fullerite...
Getting dependencies...
Linting fullerite sources...
Building fullerite...
Building beatit...
...............
----------------------------------------------------------------------
Ran 15 tests in 0.012s

OK
Testing fullerite
ok  	beatit	0.008s	coverage: 22.2% of statements
ok  	fullerite	3.015s	coverage: 39.4% of statements
ok  	fullerite/collector	7.519s	coverage: 80.8% of statements
ok  	fullerite/config	0.008s	coverage: 78.1% of statements
ok  	fullerite/handler	3.023s	coverage: 56.9% of statements
ok  	fullerite/internalserver	0.455s	coverage: 88.1% of statements
ok  	fullerite/metric	0.004s	coverage: 81.0% of statements
ok  	fullerite/util	0.038s	coverage: 93.2% of statements
bash-4.3#
```

**Comment by gnufied** [comment]
I have cleaned up the commits and merged them into master. Closing this PR. @ChristianKniep  thank you for contributing!

**Pull request closed without merging**